s3/s3.go

```diff
@@ -236,7 +245,12 @@ func (b *Bucket) PutReaderWithHeader(path string, r io.ReadSeeker, length int64,
 		return err
 	}
 	// Signing may read the request body.
-	if _, err := r.Seek(0, 0); err != nil {
+	if _, err := seeker.Seek(0, 0); err != nil {
```

Comment:
Should seek back to the position prior to signing, which is not necessarily the beginning.